wgsl/extract-grammar.py

```diff
@@ -510,7 +761,321 @@ def read_spec(options):
                     scanner_i += scanner_parse[-1]  # Advance line index
         scanner_i += 1
 
-    result[scanner_rule.name()]["_comment"] = [["`/\\/\\/.*/`"]]
+    RULE_NAME_CODE_POINTS = string.ascii_lowercase + string.digits + "_"
+    bnfdialect_data = ""
+
+    # Read the contents of the bnfdialect file
+    with open(options.syntax_filename, "r") as file:
+        bnfdialect_data = file.read()
+
+    scan_in = {"name": "blank", "value": "", "content": []}  # blank, comment, rule
+
+    syntax_dict = {}
+
+    s_i = 0
```

Comment:
This code is very detailed an confusing.

I suggest breaking the input into text lines, then use a recursive algorithm with several functions to parse it.
Being line-oriented is *much* simpler.

See my partial work here:

https://github.com/dneto0/gpuweb/blob/grammar-author/wgsl/rewrite.py#L302 

That was basically working to parse either from BS source or from plain BNF form.  The same essential structure is in both. But the code is clearer than this byte-by-byte algorithm.